### PR TITLE
Add mutex to `google_tags_tag_binding`

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes: https://github.com/hashicorp/terraform-provider-google/issues/11381

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
tags: fixed issue where tags could not be applied sequentially to the same parent in `google_tags_tag_binding`
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Drive-by comment: Is this mutex on the right parent? The error message suggests we need it on the resource, not the tag value: `Error: Error creating TagBinding: googleapi: Error 400: Resource '//cloudresourcemanager.googleapis.com/projects/XXXXXX' has another tag binding operation in progress.
with google_tags_tag_binding.bindings["fooo"],`

**Review by melinath — APPROVED** [review]
lgtm assuming tests pass and this should share with tagvalue

File: mmv1/products/tags/terraform.yaml [hunk 1]
```diff
@@ -71,6 +71,7 @@ overrides: !ruby/object:Overrides::ResourceOverrides
     autogen_async: true
     id_format: "tagBindings/{{name}}"
     import_format: ["tagBindings/{{name}}", "{{name}}"]
+    mutex: tagValues/{{parent}}
```
Comment (melinath) [inline mmv1/products/tags/terraform.yaml:74]:
```suggestion
    mutex: tagBindings/{{parent}}
```

**Review by melinath — CHANGES_REQUESTED** [review]
Okay, tags are a little weird because the resources each have a parent key and it means a different thing for each one. For tagKey parent is the organization; for tagValue the parent is a tagKey; for tagBinding the parent is the resource the tag value is being applied to. So tagBinding and tagValue would never overlap in terms of their mutex.

File: mmv1/products/tags/terraform.yaml (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/products/tags/terraform.yaml:74]: Thanks for being so thorough !

**Comment by ScottSuarez** [comment]
> Drive-by comment: Is this mutex on the right parent? The error message suggests we need it on the resource, not the tag value: `Error: Error creating TagBinding: googleapi: Error 400: Resource '//cloudresourcemanager.googleapis.com/projects/XXXXXX' has another tag binding operation in progress. with google_tags_tag_binding.bindings["fooo"],`

Should be the correct resource. This is done on the resource through the parent link supplied. Stopping sequential bindings on eq parent links should solve the issue afaik

**Comment by rileykarson** [comment]
> Should be the correct resource. This is done on the resource through the parent link supplied. Stopping sequential bindings on eq parent links should solve the issue afaik

Ah, I thought `parent` referred to the TagValue for some reason and not the linked resource. Should have double-checked. Thanks!

**Review by melinath — APPROVED** [review]
lgtm

**Comment by ScottSuarez** [comment]
API team is working to add support for this, checking this code in while they work on it b/228352372

**Pull request merged**